stdlib/camlinternalQuote.ml

```diff
@@ -18,4 +26,33 @@ struct
      Local { name; stamp = !currentstamp }
   | Global name | Predef { name ; _ } ->
      failwith ("CamlinternalQuote.rename " ^ name)
-end
+
+  let exec_in_current_scope scope f = let muted_vars = ref IdentSet.empty in match f () with 
+      effect FreeVar fvs, k-> let free = IdentSet.diff fvs (IdentSet.union (!muted_vars) scope) in 
+                             Printf.printf "AST has %d free variables \n" (IdentSet.cardinal fvs);
+                             Printf.printf "scope has %d bound variables \n" (IdentSet.cardinal scope);
+                             Printf.printf "There are %d free variables \n" (IdentSet.cardinal (IdentSet.diff fvs scope));
+                             (if not (IdentSet.is_empty free) then perform (FreeVar free));
```

Comment:
I suggest moving `check` (defined below) above the definition of `exec_in_current_scope`, then replacing this line with a call to `check`:

```suggestion
                             check free;
```